pyfr/solvers/baseadvecdiff/elements.py

```diff
@@ -25,9 +28,12 @@ def set_backend(self, backend, nscalupts, nonce, linoff):
         # Mesh regions
         regions = self._mesh_regions
 
+        comm = (self._comm_fpts if self.basis.fpts_in_upts
```

Comment:
Any reason not to assign `self._comm_fpts = self._vect_fpts.slice(0, self.nfpts)` when we decide what to allocate to avoid this check?